### PR TITLE
Use CMake config mode for gMock detection

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

File: unitTests/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-find_package(GTest REQUIRED)
+# Prefer using Google suplied CMake config as bult-in module is buggy before 3.23
+find_package(GTest REQUIRED CONFIG)
```
Comment (Ryanf55) [inline unitTests/CMakeLists.txt:2]: If gtest supplies COMPONENTS, then declare a components required. 

In the case that gtest is installed but gmock is not installed, the find_package returns found but the gmock namespace target wont exist. 

`find_package(GTest REQUIRED COMPONENTS gtest gmock gtest_main gmock_main)`

There's some churn around the target names depending on how it was found, but that's the idea.

Comment (kmilos) [reply unitTests/CMakeLists.txt]: I had a qucik look already beforehand, I don't think it does...

Comment (kmilos) [reply unitTests/CMakeLists.txt]: But there's a different problem: at least on Fedora, both `gmock` (pulled in by `gtest-devel`) and `gmock-devel` ship the shared objects (the `gmock-devel` just the unversioned symlink though), so this actually doesn't catch the missing `gmock-devel`...

**Review by neheb — APPROVED** [review]

File: ci/install_dependencies.sh [hunk 2]
```diff
@@ -24,7 +24,7 @@ debian_build_gtest() {
 centos_build_inih() {
     [-d inih_build ] || git clone https://github.com/benhoyt/inih.git inih_build
     cd inih_build
-    git checkout r56
```
Comment (neheb) [inline ci/install_dependencies.sh:27]: @eli-schwartz since the wrap is in here, can this section be replaced?

Comment (eli-schwartz) [reply ci/install_dependencies.sh:27]: I think this is also used for installing dependencies before running cmake.

Comment (neheb) [reply ci/install_dependencies.sh:27]: I asked Bard about this. It told me about an imaginary function called cmake_subproject that supported meson. And yes, it is. I was thinking mostly of replacing that git clone.

File: unitTests/CMakeLists.txt (hunk 1 above)
Comment (neheb) [reply unitTests/CMakeLists.txt]: FWIW for meson, I use gmock_main as the dependency.

**Comment by neheb** [comment]
@mergifyio backport 0.28.x

File: unitTests/CMakeLists.txt (hunk 1 above)
Comment (Ryanf55) [reply unitTests/CMakeLists.txt]: I tried using components on my branch `use-gtest-components` like so:
`find_package(GTest CONFIG COMPONENTS gtest_main gmock_main)`

On Ubuntu 22.04 with `libgmock-dev` istalled, the package found returns successfully.

If I add a bogus component `foobar`, the gtest find package call reports the component as found. Seems like a bug to me in their find module.

**Pull request merged**